src/libkomwm.py

```diff
@@ -341,34 +337,60 @@ def dump_priorities(prio_range, path, maxzoom):
+                '''
+
             prios = sorted(prio_ranges[prio_range]['priorities'].items(),
                            key = lambda item: (OVERLAYS_MAX_PRIORITY - item[1], item[0][0], item[0][1]))
             group_prio = prios[0][1]
             group = ''
+            group_comment = '# '
             for p in prios:
                 if p[1] != group_prio:
-                    outfile.write(f'{group}=== {group_prio}\n\n')
+                    if prio_range == PRIO_OVERLAYS and comment_auto_captions and group_prio < 0:
+                        for s in comment_auto_captions.splitlines():
+                            outfile.write(f'# {s.strip()}'.rstrip() + '\n')
+                        outfile.write('\n')
+                        comment_auto_captions = None
+                    outfile.write(f'{group}{group_comment}=== {group_prio}\n\n')
                     group_prio = p[1]
                     group = ''
+                    group_comment = '# '
+
                 cl = p[0][0]
                 object_id = p[0][1]
+                auto_dr_type = None
+                auto_comment = None
+                if len(p[0]) == 4:
+                    auto_dr_type = p[0][2]
+                    auto_comment = p[0][3]
 
                 line_drules = ''
                 other_drules = ''
                 if cl in visibilities:
-                    for dr_type in sorted(visibilities[cl].keys()):
-                        for oid in sorted(visibilities[cl][dr_type].keys()):
-                            dr_zoom = dr_type + oid if oid else dr_type
-                            dr_zoom += ' ' + prettify_zooms(visibilities[cl][dr_type][oid], maxzoom)
-                            # Drules matching this prio_range and object_id.
-                            if (oid == object_id and
-                                ((prio_range == PRIO_OVERLAYS and dr_type in ('icon', 'caption', 'shield', 'pathtext')) or
+                    for dr_type_comment in sorted(visibilities[cl].keys(), key = lambda drt: dr_types_order.index(drt[0])):
+                        for oid in sorted(visibilities[cl][dr_type_comment].keys()):
+                            dr_type, dr_auto_comment = dr_type_comment
+                            dr_zoom = dr_type + oid
+                            if dr_auto_comment is not None:
+                                dr_zoom = f'{dr_zoom}({dr_auto_comment})'
+                            dr_zoom += ' ' + prettify_zooms(visibilities[cl][dr_type_comment][oid], maxzoom)
+                            # Drules matching this prio_range and object_id and
+                            # - an auto priority dr_type match or
+                            # - any other non-auto dr_type suitable
+                            if (oid == object_id and (dr_type == auto_dr_type and dr_auto_comment == auto_comment or
+                                auto_dr_type is None and dr_auto_comment is None and
```

Comment:
How it should be interpreted?
```
A and B or C and D and ...
```
A and (B or C) and D
(A and B) or (C and D)
?

Please, put braces here and split this gigantic if on smaller conditions with intermediate vars.

Reply:
fixed